### PR TITLE
WCS / Clip and ship 2

### LINKED ISSUE #3109: [EPIC] WCS "clip and ship"- v2.0
From https://github.com/TerriaJS/terriajs/issues/2937#issuecomment-420085999:

* The parameters width and height determine the output grid of the resulting coverage, this means that the resulting coverage is not native resolution.
* The resulting GeoTIFF only includes the bands associated with the active band (Red, Blue, Green for this example).
* Our belief is that our stakeholders will expect to be able to acquire native resolution and all bands associated with a given product. ACTION: to validate this assumption

### PR DESCRIPTION
**Pull request by nf-s**
### What this PR does

Fixes https://github.com/terriajs/terriajs/issues/3109

#### Changes

* Re-allowed models to be added to `workbench` if the are not `Mappable` or `Chartable`
* Moved `WebMapServiceCatalogItem.GetCapbilitiesStratum` to `lib\Models\Catalog\Ows\WebMapServiceCapabilitiesStratum.ts`
* Moved `WebMapServiceCatalogItem.DiffStratum` to `DiffableMixin` (https://github.com/TerriaJS/terriajs/issues/4667)
* `callWebCoverageService` now uses version WCS `2.0.0`
  * All WCS export functionality is now in `ExportWebCoverageServiceMixin`
  * Added `WebCoverageServiceParameterTraits` to `WebMapServiceCatalogItemTraits.linkedWcsParameters`. It includes `outputFormat` and `outputCrs`
  * Will attempt to use native CRS and format (from WCS `DescribeCoverage`)
  * No longer sets `width` or `height` - so export will now return native resolution
* Fix `WebMapServiceCatalogItem.styles` if `supportsGetLegendGraphics = false`. This means that if a WMS server doesn't support `GetLegendGraphics` requests, the first style will be set as the default style.

### Todo

- [ ] ~Make `MapInteractionMode` match Phil's prototype~ for v2
- [ ] ~Add `SelectableDimension` to `MapInteractionMode`~ for v2
- [ ] ~`SelectableDimension`: Output format dropdown~ for v2
- [ ] ~`SelectableDimension`: Pixel size / native resolution~ for v2
- [ ] ~`SelectableDimension`: Band selection? (how does this work with ODC-OWS styles)~ for v2
- [x] Get WCS `GetCapabilities` and `DescribeCoverage` calls working

### Checklist

-   [x] Added tests
-   [ ] ~I've updated relevant documentation in `doc/`.~
-   [x] I've updated CHANGES.md with what I changed.

### CODE REVIEW

File: lib/Models/Catalog/Ows/WebMapServiceCapabilitiesStratum.ts [hunk 1]
```diff
@@ -0,0 +1,820 @@
...
+      ...dimensionSubsets
+    ]).map(subset => createStratumInstance(KeyValueTraits, subset));
+
+    return createStratumInstance(WebCoverageServiceParameterTraits, {
+      outputCrs,
+      subsets,
+      duplicateSubsetValues
+    });
+  }
+}
```
Comment (nf-s) [inline lib/Models/Catalog/Ows/WebMapServiceCapabilitiesStratum.ts]: This is the only new bit in this file

File: lib/Models/Definition/upsertModelFromJson.ts [hunk 2]
```diff
@@ -85,40 +85,41 @@ export default function upsertModelFromJson(
...
-    if (model.type !== StubCatalogItem.type && options.addModelToTerria) {
-      try {
+      if (model.type !== StubCatalogItem.type && options.addModelToTerria) {
         model.terria.addModel(model, json.shareKeys);
-      } catch (error) {
-        errors.push(error);
       }
+    } catch (e) {
+      errors.push(TerriaError.from(e, `Failed to create model`));
     }
```
Comment (nf-s) [inline lib/Models/Definition/upsertModelFromJson.ts:110]: Added better error message if model could not be created

File: CHANGES.md [hunk 3]
```diff
@@ -3,6 +3,16 @@ Change Log
...
+* Bigger zoom control icons.
+* Re-allowed models to be added to `workbench` if the are not `Mappable` or `Chartable`
+* Moved `WebMapServiceCatalogItem.GetCapbilitiesStratum` to `lib\Models\Catalog\Ows\WebMapServiceCapabilitiesStratum.ts`
+* Moved `WebMapServiceCatalogItem.DiffStratum` to `DiffableMixin`
+* `callWebCoverageService` now uses version WCS `2.0.0`
+  * All WCS export functionality is now in `ExportWebCoverageServiceMixin`
+  * Added `WebCoverageServiceParameterTraits` to `WebMapServiceCatalogItemTraits.linkedWcsParameters`. It includes `outputFormat` and `outputCrs`
+  * Will attempt to use native CRS and format (from `DescribeCoverage`)
+  * No longer sets `width` or `height` - so export will now return native resolution
+* `Result.throwIfUndefined()` will now only throw if `result.value` is undefined - regardless of `result.error`
```
Comment (tephenavies) [inline CHANGES.md]: I think this should be listed as a breaking change, even though it's a change to an unstable class

Comment (nf-s) [reply CHANGES.md]: 👍

**Review by tephenavies — APPROVED** [review]

**Pull request merged**